### ISSUE TITLE
Expense - Expense report is no longer an expense report after deleting expense with comment

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#42948]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.78-3
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4591742
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. Submit an expense to a user that has unsettled request
3. Click on the expense preview to go to combined report
4. Send a message
5. Delete the expense

## Expected Result:
The expense report will become empty with a total section. It will still remain as an expense report. New expense will be added to this report. In the main chat, it will show 0.00 expense preview (previous behavior)

## Actual Result:
The expense report turns to a message thread and it is no longer an expense report. In the main chat, it shows [Deleted message] instead of expense preview with total 0.00

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence
 
</details> 

https://github.com/Expensify/App/assets/78819774/0eb72486-0408-494e-a59d-6b7badd91f12

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
@abekkala  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by abekkala** [comment]
hmm, I don't have access to ND right now and unable to reproduce. But, are we certain this is expected? @alexpensify as the curator of this wave can you confirm the expectation here? 

### Expected Result:
The expense report will become empty with a total section. It will still remain as an expense report. New expense will be added to this report. In the main chat, it will show 0.00 expense preview (previous behavior)

**Comment by trjExpensify** [comment]
It's super hard to follow this issue. 

> 1. Go to staging.new.expensify.com
> 2. Submit an expense to a user that has unsettled request
> 3. Click on the expense preview to go to combined report
> 4. Send a message
> 5. Delete the expense

Step 2 suggests the user adds an expense to an existing expense report that has another expense on it, then deletes only one of those expenses in step 5. So nah, we wouldn't delete the whole expense report here without a trace. The one expense that was deleted would appear on the expense report as `[Deleted expense]` to preserve the expense comment thread on said report.

(Think of this a bit like Slack and deleting threads/message). 

1. If all expenses on an expense report are deleted, the expense report is deleted without a trace. 
2. If all expenses on a report are deleted and there are comments in the expense report to preserve, the expense report is converted to a chat report just to preserve the comments, and remains as `[Deleted report]`.
